### ISSUE TITLE
Pydantic serializer warnings when dumping a completion response with logprobs

### ISSUE THREAD

**Issue by kddubey** [openai/openai-python#744]
### Confirm this is an issue with the Python library and not an underlying OpenAI API

- [X] This is an issue with the Python library

### Describe the bug

Dumping a completion response from a request with `logprobs` set raises this warning:

```
python3.11/site-packages/pydantic/main.py:308: UserWarning: Pydantic serializer warnings:
  Expected `int` but got `float` - serialized value may not be as expected
  Expected `int` but got `float` - serialized value may not be as expected
  Expected `int` but got `float` - serialized value may not be as expected
  return self.__pydantic_serializer__.to_python(
```

I confirmed that changing [this type annotation](https://github.com/openai/openai-python/blob/ba424470504a470d1b0902847bcfae7149deae33/src/openai/types/completion_choice.py#L18) to

```python
    top_logprobs: Optional[List[Dict[str, float]]] = None
```

removes the warnings.

### To Reproduce

1. Install `openai>=1.0.0`
2. In a `completions` API call, set `logprobs>=1`
3. Dump the response using `.model_dump()`, as advised in the [migration guide](https://github.com/openai/openai-python/discussions/742)

### Code snippets

```Python
import openai

client = openai.OpenAI()

# Set logprobs >= 1
response = client.completions.create(
    model="curie", prompt="test", max_tokens=3, logprobs=1
)

response_as_dict = response.model_dump()
# raises a UserWarning
```

To ignore the warnings, the user currently has to do this:

```Python
import warnings

with warnings.catch_warnings():
    warnings.filterwarnings(
        "ignore",
        category=UserWarning,
        message="Pydantic serializer warnings",
    )
    response_as_dict = response.model_dump()
```

### OS

macOS

### Python version

Python v3.11.5

### Library version

openai v1.1.2

**Comment by RobertCraigie** [comment]
Thanks for the report, this will be fixed in the next release. https://github.com/openai/openai-python/pull/738/files#diff-1d262046f5d37b8e4ee402833a96289d8264757a8f6b69bb843170aa295ca0cd

**Comment by RobertCraigie** [comment]
This is fixed in `v1.2.0`!
